### ISSUE TITLE
Warnings with `dotnet build` 

### ISSUE THREAD

**Issue by thygrrr** [DaveGreen-Games/ShapeEngine#28]
## What:
Checked-out as-is, dotnet bluild will raise a variety of warnings, including some mildly concerning ones (probably from nullable/not nullable friction).

## How:
```powershell
git clone git@github.com:DaveGreen-Games/ShapeEngine.git
cd ShapeEngine
dotnet build
```

then...

```powershell
MSBuild version 17.9.8+b34f75857 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
P:\ShapeEngine\ShapeEngine\Input\InputAction.cs(262,47): warning CS8604: Possible null reference argument for parameter 'gamepad' in 'InputAction.InputAction(uint accessTag, ShapeGamepadDevice gamepad, params IInputType[] inputTypes)'. [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\ShapeEngine\UI\ControlNodeContainer.cs(133,43): warning CS8602: Dereference of a possibly null reference. [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\ShapeEngine\UI\ControlNodeContainer.cs(359,25): warning CS8602: Dereference of a possibly null reference. [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\ShapeEngine\Core\Game.cs(126,21): warning CS0414: The field 'Game.lastControlledMousePosition' is assigned but its value is never used [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\ShapeEngine\Core\Game.cs(127,18): warning CS0414: The field 'Game.mouseControlled' is assigned but its value is never used [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
  ShapeEngine -> P:\ShapeEngine\ShapeEngine\bin\Debug\net7.0\ShapeEngine.dll
P:\ShapeEngine\Examples\Scenes\ExampleScenes\EndlessSpaceCollision.cs(583,30): warning CS0108: 'EndlessSpaceCollision.Ship.OnKilled' hides inherited member 'GameObject.OnKilled'. Use the new keyword if hiding was intended. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\UIElements\ExampleSelectionButton.cs(114,21): warning CS0162: Unreachable code detected [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\DataExample.cs(220,12): warning CS8618: Non-nullable field 'planet' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUICancel' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIBack' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIAccept' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIAcceptMouse' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUILeft' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIRight' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIUp' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIDown' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUINextTab' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIPrevTab' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUINextPage' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIPrevPage' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionFullscreen' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionMaximize' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionMinimize' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionNextMonitor' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionCRTMinus' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionCRTPlus' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionZoom' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionCyclePalette' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionReset' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable field 'fpsLabel' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable field 'paletteInfoBox' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ControlNodeExampleScene.cs(228,16): warning CS8618: Non-nullable field 'optionButton' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\EndlessSpaceCollision.cs(625,16): warning CS8618: Non-nullable field 'iaMoveHor' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\EndlessSpaceCollision.cs(625,16): warning CS8618: Non-nullable field 'iaMoveVer' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\EndlessSpaceCollision.cs(1913,23): warning CS8604: Possible null reference argument for parameter 'collisionHandler' in 'Autogun.Autogun(CollisionHandler collisionHandler, AutogunStats stats, BulletStats bulletStats, PaletteColor color)'. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\PathfinderExample2.cs(60,16): warning CS8618: Non-nullable field 'iaMoveHor' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\PathfinderExample2.cs(60,16): warning CS8618: Non-nullable field 'iaMoveVer' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\PolygonHolesExample.cs(103,12): warning CS8618: Non-nullable field 'curPolygon' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\PathfinderExample.cs(114,16): warning CS8618: Non-nullable field 'endFlag' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(161,16): warning CS8618: Non-nullable field 'iaMoveHor' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(161,16): warning CS8618: Non-nullable field 'iaMoveVer' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(166,16): warning CS8618: Non-nullable field 'iaMoveHor' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(166,16): warning CS8618: Non-nullable field 'iaMoveVer' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ControlNodeExampleScene.cs(202,44): warning CS0169: The field 'ControlNodeExampleScene.optionButton' is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\InputExample.cs(857,25): warning CS0414: The field 'InputExample.lastMouseDelta' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(287,22): warning CS0414: The field 'ScreenEffectsExample.sliderFill' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\InputExample.cs(858,25): warning CS0414: The field 'InputExample.lastMouseWheel' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(286,22): warning CS0414: The field 'ScreenEffectsExample.slider' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\CameraAreaDrawExample.cs(16,17): warning CS0414: The field 'CameraAreaDrawExample.movementDir' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
  Examples -> P:\ShapeEngine\Examples\bin\Debug\net7.0\Examples.dll

Build succeeded.

P:\ShapeEngine\ShapeEngine\Input\InputAction.cs(262,47): warning CS8604: Possible null reference argument for parameter 'gamepad' in 'InputAction.InputAction(uint accessTag, ShapeGamepadDevice gamepad, params IInputType[] inputTypes)'. [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\ShapeEngine\UI\ControlNodeContainer.cs(133,43): warning CS8602: Dereference of a possibly null reference. [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\ShapeEngine\UI\ControlNodeContainer.cs(359,25): warning CS8602: Dereference of a possibly null reference. [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\ShapeEngine\Core\Game.cs(126,21): warning CS0414: The field 'Game.lastControlledMousePosition' is assigned but its value is never used [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\ShapeEngine\Core\Game.cs(127,18): warning CS0414: The field 'Game.mouseControlled' is assigned but its value is never used [P:\ShapeEngine\ShapeEngine\ShapeEngine.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\EndlessSpaceCollision.cs(583,30): warning CS0108: 'EndlessSpaceCollision.Ship.OnKilled' hides inherited member 'GameObject.OnKilled'. Use the new keyword if hiding was intended. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\UIElements\ExampleSelectionButton.cs(114,21): warning CS0162: Unreachable code detected [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\DataExample.cs(220,12): warning CS8618: Non-nullable field 'planet' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUICancel' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIBack' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIAccept' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIAcceptMouse' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUILeft' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIRight' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIUp' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIDown' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUINextTab' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIPrevTab' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUINextPage' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionUIPrevPage' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionFullscreen' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionMaximize' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionMinimize' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionNextMonitor' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionCRTMinus' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionCRTPlus' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionZoom' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionCyclePalette' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable property 'InputActionReset' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable field 'fpsLabel' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\GameloopExamples.cs(264,16): warning CS8618: Non-nullable field 'paletteInfoBox' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ControlNodeExampleScene.cs(228,16): warning CS8618: Non-nullable field 'optionButton' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\EndlessSpaceCollision.cs(625,16): warning CS8618: Non-nullable field 'iaMoveHor' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\EndlessSpaceCollision.cs(625,16): warning CS8618: Non-nullable field 'iaMoveVer' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\EndlessSpaceCollision.cs(1913,23): warning CS8604: Possible null reference argument for parameter 'collisionHandler' in 'Autogun.Autogun(CollisionHandler collisionHandler, AutogunStats stats, BulletStats bulletStats, PaletteColor color)'. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\PathfinderExample2.cs(60,16): warning CS8618: Non-nullable field 'iaMoveHor' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\PathfinderExample2.cs(60,16): warning CS8618: Non-nullable field 'iaMoveVer' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\PolygonHolesExample.cs(103,12): warning CS8618: Non-nullable field 'curPolygon' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\PathfinderExample.cs(114,16): warning CS8618: Non-nullable field 'endFlag' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(161,16): warning CS8618: Non-nullable field 'iaMoveHor' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(161,16): warning CS8618: Non-nullable field 'iaMoveVer' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(166,16): warning CS8618: Non-nullable field 'iaMoveHor' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(166,16): warning CS8618: Non-nullable field 'iaMoveVer' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ControlNodeExampleScene.cs(202,44): warning CS0169: The field 'ControlNodeExampleScene.optionButton' is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\InputExample.cs(857,25): warning CS0414: The field 'InputExample.lastMouseDelta' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(287,22): warning CS0414: The field 'ScreenEffectsExample.sliderFill' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\InputExample.cs(858,25): warning CS0414: The field 'InputExample.lastMouseWheel' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\ScreenEffectsExample.cs(286,22): warning CS0414: The field 'ScreenEffectsExample.slider' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
P:\ShapeEngine\Examples\Scenes\ExampleScenes\CameraAreaDrawExample.cs(16,17): warning CS0414: The field 'CameraAreaDrawExample.movementDir' is assigned but its value is never used [P:\ShapeEngine\Examples\Examples.csproj]
    49 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.18
```

**Comment by SoloByte** [comment]
I know about this warnings but I don´t care too much about them :) Most of them I think are values that get initialized in a function that is called in the constructor ( which leads to a warning even though it is initialized correctly). 
There may be some parts were I did things the not-so-correct way but I don´t mind for now. 
The only warnings I am going to investigate are the ones in the ShapeEngine project. 

Once I have resolved them by fixing or ignoring I will close this issue!
